File: frontend-mobile/biblio_upb/lib/models/catalogue.dart
class Catalogue {
  final int? catalogueId;
  final int numero;
  final String titre;
  final String categorie;
  final String description;
  final String icone;
  final String couleur;

  Catalogue({
    this.catalogueId,
    required this.numero,
    required this.titre,
    required this.categorie,
    required this.description,
    required this.icone,
    required this.couleur,
  });

  factory Catalogue.fromJson(Map<String, dynamic> json) => Catalogue(
    catalogueId: json['catalogueId'],
    numero: json['numero'] ?? 0,
    titre: json['titre'] ?? '',
    categorie: json['categorie'] ?? '',
    description: json['description'] ?? '',
    icone: json['icone'] ?? 'Package',
    couleur: json['couleur'] ?? 'blue',
  );
}

File: backend/bibliotheque/src/main/java/com/upb/bibliotheque/entity/Catalogue.java
package com.upb.bibliotheque.entity;

import jakarta.persistence.*;
import lombok.*;
import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.List;

@Entity
@Table(name = "catalogues")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class Catalogue {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long catalogueId;

    private Integer numero;
    @Column(nullable = false)
    private String titre;
    private String categorie;
    private String description;
    private String icone;
    private String couleur;

    @OneToMany(mappedBy = "catalogue", cascade = CascadeType.ALL)
    @JsonIgnore
    private List<Livre> livres;
}